network/config_test.go

```diff
@@ -0,0 +1,193 @@
+        t.Fatalf("Expected to set book to %v. Actual: %v", book, config.book)
+    }
+}
+
+func TestSetCodec(t *testing.T) {
+    config := DefaultConfig
+    codec := DummyCodec{}
+    
+    setFunc := SetCodec(codec)
+    setFunc(&config)
+
+    if config.codec != codec {
+        t.Fatalf("Expected to set codec to %v. Actual: %v", codec, config.codec)
+    }
+}
+
+func TestSetDiscovery(t *testing.T) {
+    config := DefaultConfig
+    discovery := time.Duration(5)
+    
+    setFunc := SetDiscovery(discovery)
+    setFunc(&config)
+
+    if config.discovery != discovery {
+        t.Fatalf("Expected to set discovery to %s. Actual: %s", discovery, config.discovery)
+    }
+}
+
+func TestSetHeartbeat(t *testing.T) {
+    config := DefaultConfig
+    heartbeat := time.Duration(5)
+    
+    setFunc := SetHeartbeat(heartbeat)
+    setFunc(&config)
+
+    if config.heartbeat != heartbeat {
+        t.Fatalf("Expected to set heartbeat to %s. Actual: %s", heartbeat, config.heartbeat)
+    }
+}
+
+func TestSetLog(t *testing.T) {
+    config := DefaultConfig
+    log, _ := zap.NewDevelopment()
+    
+    setFunc := SetLog(log)
+    setFunc(&config)
+
+    if config.log != log {
+        t.Fatalf("Expected to set log to %v. Actual: %v", log, config.log)
+    }
+}
+
+func TestSetMaxPeers(t *testing.T) {
+    config := DefaultConfig
+    maxPeers := uint(15)
+    
+    setFunc := SetMaxPeers(maxPeers)
+    setFunc(&config)
+
+    if config.maxPeers != maxPeers {
+        t.Fatalf("Expected to set maxPeers to %d. Actual: %d", maxPeers, config.maxPeers)
+    }
+}
+
+func TestSetMinPeers(t *testing.T) {
+    config := DefaultConfig
+    minPeers := uint(5)
+    
+    setFunc := SetMinPeers(minPeers)
+    setFunc(&config)
+
+    if config.minPeers != minPeers {
+        t.Fatalf("Expected to set minPeers to %d. Actual: %d", minPeers, config.minPeers)
+    }
+}
+
+func TestSetNetwork(t *testing.T) {
+    config := DefaultConfig
+    network := make([]byte, 2)
+    network[0] = 5
+    network[1] = 10
+    
+    setFunc := SetNetwork(network)
+    setFunc(&config)
+
+    if config.network[0] != network[0] || config.network[1] != network[1] {
+        t.Fatalf("Expected to set network to %d. Actual: %d", network, config.network)
+    }
+}
+
+func TestSetServer(t *testing.T) {
+    config := DefaultConfig
+    server := true
+    
+    setFunc := SetServer(server)
+    setFunc(&config)
+
+    if config.server != server {
+        t.Fatalf("Expected to set server to %v. Actual: %v", server, config.server)
+    }
+}
+
+func TestSetSubscriber(t *testing.T) {
+    config := DefaultConfig
+    subscriber := make(chan interface{})
+    
+    setFunc := SetSubscriber(subscriber)
+    setFunc(&config)
+
+    if config.subscriber != subscriber {
+        t.Fatalf("Expected to set subscriber to %v. Actual: %v", subscriber, config.subscriber)
+    }
+}
+
+func TestSetTimeout(t *testing.T) {
+    config := DefaultConfig
+    timeout := time.Duration(5)
```

Comment:
Yea, I was mostly focused on checking the end result, not values we specify. Going to change it however.